scm/github.go

```diff
@@ -335,16 +327,7 @@ func (s *GithubSCM) CreateHook(ctx context.Context, opt *CreateHookOptions) erro
 		},
 		Events: []string{"push", "pull_request", "pull_request_review"},
 	}
-	var err error
-	// prioritize creating an organization hook
-	if opt.Organization != "" {
-		_, _, err = s.client.Organizations.CreateHook(ctx, opt.Organization, hook)
-		if err != nil {
-			return fmt.Errorf("CreateOrgHook: failed to create GitHub hook for org %s: %w", opt.Organization, err)
-		}
-	} else {
-		_, _, err = s.client.Repositories.CreateHook(ctx, opt.Repository.Owner, opt.Repository.Path, hook)
```

Comment:
As we no longer use `opt.Repository` for GitHub we can remove it from `TestCreateHook` in `github_test.go`

```go
// github_test.go line 92:96
opt := &scm.CreateHookOptions{
	URL:        serverURL,
	Secret:     secret,
	Repository: &scm.Repository{Owner: qfTestOrg, Path: "tests"},
}
```
and instead use `opt.Organization`. This way `opt.Valid()` (in the context of this specific test) will return `true` due to `opt.Organization != ""` rather than the repository part of the check:

```go
opt.URL != "" &&
		(opt.Organization != "" || (opt.Repository != nil &&
			opt.Repository.valid()))
```

Reply:
Let's remove dependencies on gitlab.